src/webviews/extension-side/dataviewer/interpreterDataViewerDependencyImplementation.node.ts

```diff
@@ -0,0 +1,125 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { SemVer } from 'semver';
+import { CancellationToken, CancellationTokenSource } from 'vscode';
+import { ProductNames } from '../../../kernels/installer/productNames';
+import { IInstaller, Product, InstallerResponse } from '../../../kernels/installer/types';
+import { IApplicationShell } from '../../../platform/common/application/types';
+import { Cancellation, createPromiseFromCancellation } from '../../../platform/common/cancellation';
+import { traceWarning } from '../../../platform/logging';
+import { IPythonExecutionFactory } from '../../../platform/common/process/types.node';
+import { parseSemVer } from '../../../platform/common/utils';
+import { DataScience } from '../../../platform/common/utils/localize';
+import { IInterpreterService } from '../../../platform/interpreter/contracts';
+import { PythonEnvironment } from '../../../platform/pythonEnvironments/info';
+import { sendTelemetryEvent, Telemetry } from '../../../telemetry';
+import { pandasMinimumVersionSupportedByVariableViewer } from './constants';
+import { BaseDataViewerDependencyImplementation } from './baseDataViewerDependencyImplementation';
+
+/**
+ * Uses the Python interpreter to manage dependencies of a Data Viewer.
+ */
+export class InterpreterDataViewerDependencyImplementation extends BaseDataViewerDependencyImplementation {
+    constructor(
+        private readonly installer: IInstaller,
+        private pythonFactory: IPythonExecutionFactory,
+        private interpreterService: IInterpreterService,
+        applicationShell: IApplicationShell,
+        isCodeSpace: boolean
+    ) {
+        super(applicationShell, isCodeSpace);
+    }
+
+    public async checkAndInstallMissingDependencies(interpreter: PythonEnvironment): Promise<void> {
+        sendTelemetryEvent(Telemetry.DataViewerUsingInterpreter);
+
+        const tokenSource = new CancellationTokenSource();
+        try {
+            const pandasVersion = await this.getVersion(interpreter, tokenSource.token);
+
+            if (Cancellation.isCanceled(tokenSource.token)) {
+                return;
+            }
+
+            if (pandasVersion) {
+                if (pandasVersion.compare(pandasMinimumVersionSupportedByVariableViewer) > 0) {
+                    return;
+                }
+                sendTelemetryEvent(Telemetry.PandasTooOld);
+                // Warn user that we cannot start because pandas is too old.
+                const versionStr = `${pandasVersion.major}.${pandasVersion.minor}.${pandasVersion.build}`;
+                throw new Error(DataScience.pandasTooOldForViewingFormat().format(versionStr));
```

Comment:
Shouldn't the prompt be displayed at this point in time?
I.e. if the version is too old, shouldn't we display the prmonpt and ask to install.